nix/packages/tool.nix

```diff
@@ -0,0 +1,54 @@
+{ stdenv
+, systemd
+, binutils-unwrapped
+, sbsigntool
+, rustPlatform
+, lib
+, runCommand
+, fetchurl
+, clippy
+, path
+, enableLint ? false
+}:
+rustPlatform.buildRustPackage
+  {
+    pname = "lanzatool";
+    version = "0.3.0";
+    src = runCommand "src" { } ''
+      install -D ${../../rust/tool/Cargo.toml} $out/Cargo.toml
+      install -D ${../../rust/tool/Cargo.lock} $out/Cargo.lock
+      cp -r ${../../rust/tool/src} $out/src
+    '';
+
+    TEST_SYSTEMD = systemd;
+
+    nativeBuildInputs = lib.optional enableLint clippy;
+
+    cargoLock = {
+      lockFile = ../../rust/tool/Cargo.lock;
+    };
+
+    nativeCheckInputs = [
+      binutils-unwrapped
+      sbsigntool
+    ];
+
+    meta = with lib; {
+      description = "Lanzaboote UEFI tooling for SecureBoot enablement on NixOS systems";
+      homepage = "https://github.com/nix-community/lanzaboote";
+      license = licenses.mit;
+    };
+  } // lib.optionalAttrs enableLint {
```

Comment:
This fells really janky, but I guess there is not really a better way. Can you also add something like this for `cargo fmt`?

Reply:
actually I find it really neat, we *don't need* to build the project to run clippy or cargo fmt.
Of course, we can be more fancy and extract the drvs, etc and put a proper checkPhase for that, but that seems wasteful?

Reply:
Nono, please let's not make this more complicated than it needs to be. I kind of just wished that there was a flag you can just enable in `buildRustPackage`. 

> we _don't need_ to build the project to run clippy

True, but running clippy is almost like building the project. 

In any case, I think your solution is good for what we can do right here downstream, but I just wished there was better support for these things upstream (or I guess for development `crane` really kind of is just better).

Reply:
Feel free to go upstream and propose stuff :P

Reply:
pls no piling more stuff into buildRustPackage — it doesn't compose and makes Cargo special compared to other build systems for no reason.  I'd be okay with a clippy setup hook though.

Reply:
Why not have it in passthru.lint and build $NAME.lint?

Reply:
I cannot do this if I don't have any recursive reference to myself, which is not possible because I don't have finalAttrs, neither a proper callPackage reinjector AFAIK.

Reply:
@nikstur Added something for cargo fmt, let me know if this is not enough.